modules/widgets/src/zoom-widget.tsx

```diff
@@ -8,7 +8,7 @@ import type {Deck, Viewport, Widget, WidgetPlacement} from '@deck.gl/core';
 import {render} from 'preact';
 import {ButtonGroup, GroupedIconButton} from './components';
 
-interface ZoomWidgetProps {
+export interface ZoomWidgetProps {
```

Comment:
@Pessimistress I'm noticing that the widget's `onViewportChange` isn't called on initialization. The user needs to interact with deck before a viewport is set.

Reply:
The cause is this [guard](https://github.com/visgl/deck.gl/blob/e9eada678c485520e0e63cff9436230ae7d34fa8/modules/core/src/lib/widget-manager.ts#L263) is hit before the viewports have been assigned to the widget. `lastViewports` was set on the first time props are set (when `widgets: []` and it initializes the tooltip). 
<img width="1728" alt="Screenshot 2024-12-12 at 5 18 07 PM" src="https://github.com/user-attachments/assets/604c1166-3d57-4225-8a04-843e32d70d44" />

The goal of this code is to cache the current viewport so that the widget can use it as the starting point of its view modification.

The hook adds widgets in one-by-one: `[]` then `[FirstWidget]` then [FirstWidget, SecondWidget]`

Reply:
I guess we could push this check down into each widget to implement since the manager isn't aware of how widgets store viewports, or we could add `viewports: {}` to `class Widget` and have the manager check that instead? Thoughts?

Reply:
See proposed fix in #9303